TYPO3.Flow/Classes/TYPO3/Flow/Composer/Utility.php

```diff
@@ -0,0 +1,172 @@
+ * It is free software; you can redistribute it and/or modify it under    *
+ * the terms of the GNU Lesser General Public License, either version 3   *
+ * of the License, or (at your option) any later version.                 *
+ *                                                                        *
+ * The TYPO3 project - inspiring people to share!                         *
+ *                                                                        */
+
+use TYPO3\Flow\Package\PackageInterface;
+use TYPO3\Flow\Reflection\ObjectAccess;
+use TYPO3\Flow\Utility\Files;
+
+/**
+ * Utility to access composer information like composer manifests (composer.json) and the lock file.
+ */
+class Utility
+{
+    /**
+     * Runtime cache for composer.json data
+     *
+     * @var array
+     */
+    protected static $composerManifestCache;
+
+    /**
+     * Runtime cache for composer.lock data
+     *
+     * @var array
+     */
+    protected static $composerLockCache;
+
+    /**
+     * Returns contents of Composer manifest - or part there of.
+     *
+     * @param string $manifestPath
+     * @param string $configurationPath Optional. Only return the part of the manifest indexed by configurationPath
+     * @return array|mixed
+     */
+    public static function getComposerManifest($manifestPath, $configurationPath = null)
+    {
+        $composerManifest = static::readComposerManifest($manifestPath);
+        if ($composerManifest === null) {
+            return null;
+        }
+
+        if ($configurationPath !== null) {
+            return ObjectAccess::getPropertyPath($composerManifest, $configurationPath);
+        } else {
+            return $composerManifest;
+        }
+    }
+
+    /**
+     * Read the content of the composer.lock
+     *
+     * @return array
+     */
+    public static function readComposerLock()
+    {
+        if (self::$composerLockCache === null) {
+            if (!file_exists(FLOW_PATH_ROOT . 'composer.lock')) {
+                return [];
+            }
+            $json = file_get_contents(FLOW_PATH_ROOT . 'composer.lock');
+            $composerLock = json_decode($json, true);
+            $composerPackageVersions = isset($composerLock['packages']) ? $composerLock['packages'] : array();
+            $composerPackageDevVersions = isset($composerLock['packages-dev']) ? $composerLock['packages-dev'] : array();
+            self::$composerLockCache = array_merge($composerPackageVersions, $composerPackageDevVersions);
+        }
+
+        return self::$composerLockCache;
+    }
+
+    /**
+     * Read the content of composer.json in the given path
+     *
+     * @param string $manifestPath
+     * @return array
+     * @throws Exception\MissingPackageManifestException
+     */
+    protected static function readComposerManifest($manifestPath)
+    {
+        $manifestPathAndFilename = $manifestPath . 'composer.json';
+        if (isset(self::$composerManifestCache[$manifestPathAndFilename])) {
```

Comment:
Can we somehow streamline the code for the cache check? Here it's quick return and above the loading logic is wrapped in the if